File: backend/training-session-service/src/main/java/com/ust/excelsports/dto/TrainingHistoryResponseDto.java
package com.ust.excelsports.dto;

import lombok.Data;

import java.time.LocalDate;
import java.util.List;
import java.util.Map;

@Data
public class TrainingHistoryResponseDto {
    private int sessionId;
    private LocalDate date;
    private int duration;
    private int athleteId;
    private int totalCaloriesBurned;  // Total calories burned in the session
    private Map<String, List<ExerciseSummary>> categorizedExercises; // Exercises categorized by type

    @Data
    public static class ExerciseSummary {
        private String exerciseName;
        private int currentValue;
        private double caloriesBurned;  // Calories burned for the specific exercise
    }
}

File: front-end/excelsports/src/model/TrainingHistoryResponseDto.ts
export class TrainingHistoryResponseDto {
    sessionId: number;
    date: string;
    duration: number;
    athleteId: number;
    totalCaloriesBurned: number;
    categorizedExercises: Map<string, ExerciseSummary[]>; 
  
    constructor(
      sessionId: number,
      date: string,
      duration: number,
      athleteId: number,
      totalCaloriesBurned: number,
      categorizedExercises: Map<string, ExerciseSummary[]>
    ) {
      this.sessionId = sessionId;
      this.date = date;
      this.duration = duration;
      this.athleteId = athleteId;
      this.totalCaloriesBurned = totalCaloriesBurned;
      this.categorizedExercises = categorizedExercises;
    }
  }
  
  export class ExerciseSummary {
    exerciseName: string;
    currentValue: number;
    caloriesBurned: number;  
  
    constructor(exerciseName: string, currentValue: number, caloriesBurned: number) {
      this.exerciseName = exerciseName;
      this.currentValue = currentValue;
      this.caloriesBurned = caloriesBurned;
    }
  }
  